components/script/dom/bindings/structuredclone.rs

```diff
@@ -104,14 +104,21 @@ impl StructuredCloneReader {
 }
 
 unsafe fn read_blob(cx: *mut JSContext,
-                    r: *mut JSStructuredCloneReader)
+                    r: *mut JSStructuredCloneReader,
+                    closure: *mut raw::c_void)
```

Comment:
Let's make this take `&mut StructuredCloneHolder` instead. Then we don't need `with_structured_clone_holder` and can use `holder.blob = Some(blob)` directly.